src/rdkafka_cgrp.c

```diff
@@ -912,11 +925,12 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                      member_cnt,
                      ErrorCode ? rd_kafka_err2str(ErrorCode) : "(no error)");
 
+
+
         if (!ErrorCode) {
-                char *my_member_id;
```

Comment:
It's also used in the event we receive a `RD_KAFKA_RESP_ERR_MEMBER_ID_REQUIRED`. Although I see you mention I should use a local variable there as well. Is there a particular reason for doing it this way rather than placing it in the larger scope.

Reply:
Yes, logical scoping, it is a temp variable that only has relevance where it is used.
Having it locally scoped helps the reader of the code know what is relevant where.